VocaDbWeb/wwwroot/Scripts/Shared/EntryUrlMapper.ts

```diff
@@ -69,6 +70,10 @@ import TagBaseContract from '../DataContracts/Tag/TagBaseContract';
 			return EntryUrlMapper.details(EntryType.Song, entry.id, entry.urlFriendlyName);
 		}
 
+		public static details_artist(entry: ArtistContract) {
+			return EntryUrlMapper.details(EntryType.Artist, entry.id, entry.name);

```

Comment:
@riipah Does it make sense to specify `entry.name` as the third parameter (`userFriendlyName` or `slug`) of the `EntryUrlMapper.details` method? Or using `EntryUrlMapper.details(EntryType.Artist, entry.id)` instead of `EntryUrlMapper.details(EntryType.Artist, entry.id, entry.name)` is enough?

Reply:
You can't use entry.name. Artist name may contain characters that are not valid for URL. At the very least it would need to be URL encoded, or incompatible characters be removed. But even then you'd still have the problem that the friendly name (slug) is different depending on user's language. If you check what UrlFriendlyNameFactory is doing in server-side, it's doing quite a lot more. The task for implementing friendly names for all objects is #9